lib/queue.js

```diff
@@ -1,58 +1,216 @@
+      options.delay || 0,
+      'priority', options.priority || 0,
+      'tags', JSON.stringify(options.tags || []),
+      'retries', options.retries || 5,
+      'depends', JSON.stringify(options.depends || []));
   }
 
-  pop(cb) {
-    this.multipop(null, cb);
+  /**
+   * Make a recurring job
+   *
+   * The following attributes are required:
+   *  - klass
+   *  - interval (seconds between jobs)
+   *
+   * The following attributes are optional:
+   *  - jid (defaults to a uuid)
+   *  - data (defaults to {}, must be JSON-serializable)
+   *  - offset (delay in seconds for first job)
+   *  - priority (higher is more urgent)
+   *  - tags (array of strings)
+   *  - retries (how many times each job can be retried)
+   */
+  recur(options) {
+    // TODO(dan): should this yell if you provide an unsupported option?
+    // Suppose you set options.prority [sic] -- it should yell at you for
+    // misspelling "priority."
+    return this.client.call(
+      'recur',
+      this.name,
+      options.jid || Queue.makeJobId(),
+      options.klass,
+      JSON.stringify(options.data || {}),
+      'interval', options.interval, options.offset || 0,
+      'priority', options.priority || 0,
+      'tags', JSON.stringify(options.tags || []),
+      'retries', options.retries || 5);
   }
 
-  multipop(count, cb) {
-    this.client.call('pop', this.name, this.workerName, count || 1, makeCb(cb, jids => {
-      const jobs = _.map(JSON.parse(jids), j => new job.Job(this.client, j));
-      return cb(null, count ? jobs : (jobs[0] || null));
-    }));
+  /**
+   * Pop a job from the queue.
+   *
+   * If a count is provided, at most `count` jobs will be returned in an array. If
+   * no count is provided, then either a job will be returned or null.
+   */
+  pop(count) {
+    count = count || 1;
+    return this.client.call('pop', this.name, this.client.workerName, count)
+      .then(JSON.parse)
+      .then((response) => {
+        if (count > 1) {
+          return response.map(data => new Job(this.client, data));
+        } else if (response.length) {
+          return new Job(this.client, response[0]);
+        } else {
+          return null;
+        }
+      });
   }
+
+  /**
+   * Peek at the next job in the queue.
+   *
+   * If a count is provided, at most `count` jobs will be returned in an array. If
+   * no count is provided, then either a job will be returned or null.
+   */
+  peek(count) {
+    count = count || 1;
+    return this.client.call('peek', this.name, count)
+      .then(JSON.parse)
+      .then((response) => {
+        if (count > 1) {
+          return response.map(data => new Job(this.client, data));
+        } else if (response.length) {
+          return new Job(this.client, response[0]);
+        } else {
+          return null;
+        }
+      });
+  }
+
+  /**
+   * Return the current statistics for a given queue on a given date.
+   * The results are returned are a JSON blob::
```

Comment:
s/The results are returned are a JSON blob::/The results returned are a JSON blob:/